pytest-dev/pytest
# Merge pytest-catchlog plugin

In this PR the pytest-catchlog plugin is merged (develop branch of the
pytest-catchlog repo) into pytest-core, as
suggested in (eisensheng/pytest-catchlog#8). In the course of that I've renamed the
plugin from CatchLogPlugin to LoggingPlugin (see eisensheng/pytest-catchlog#8).

pytest-catchlog contains a backward compatibilty code for pytest-capturelog,
which I removed in this PR, since I don't think that it makes sense to merge 
this code into pytest-core.  Futhermore, the pytest-catchlog repo contains
performance tests
(https://github.com/eisensheng/pytest-catchlog/tree/develop/tests/perf)
which are not yet part of this PR. Do we want to integrate them? If
yes, how?

The documentation for this PR still needs to be integrated into this
repo. I'll probably copy the docu from the readme of pytest-catchlog and
adapt it a bit.

  * name it `$issue_id.$type` for example (588.bug)
  * if you don't have an issue_id change it to the pr id after creating the pr
  * ensure type is one of `removal`, `feature`, `bugfix`, `vendor`, `doc` or `trivial`
  * Make sure to use full sentences with correct case and punctuation, for example: "Fix issue with non-ascii contents in doctest text files."

Unless your change is a trivial or a documentation fix (e.g.,  a typo or reword of a small section) please:
